packages/extension/src/components/pages/SignTransaction/SignTransaction.tsx

```diff
@@ -0,0 +1,228 @@
+import { FC, useCallback, useEffect, useState } from 'react';
+
+import ErrorIcon from '@mui/icons-material/Error';
+import { Button, Container, Paper, Typography } from '@mui/material';
+import ReactJson from 'react-json-view';
+import { Transaction } from 'xrpl';
+
+import {
+  GEM_WALLET,
+  ReceiveSignTransactionBackgroundMessage,
+  ResponseType
+} from '@gemwallet/constants';
+
+import { ERROR_RED, SECONDARY_GRAY } from '../../../constants';
+import { useLedger, useNetwork } from '../../../contexts';
+import { useFees, useTransactionStatus } from '../../../hooks';
+import { TransactionStatus } from '../../../types';
+import { parseTransactionParam } from '../../../utils';
+import { serializeError } from '../../../utils/errors';
+import { Fee } from '../../organisms';
+import DisplayXRPLTransaction from '../../organisms/DisplayXRPLTransaction/DisplayXRPLTransaction';
+import { PageWithTitle } from '../../templates';
+
+interface Params {
+  id: number;
+  // SignTransaction fields
+  txParam: Transaction | null;
+}
+
+export const SignTransaction: FC = () => {
+  const [params, setParams] = useState<Params>({
+    id: 0,
+    // SignTransaction fields
+    txParam: null
+  });
+  const [errorRequestRejection, setErrorRequestRejection] = useState<Error>();
+  const [isParamsMissing, setIsParamsMissing] = useState(false);
+  const [transaction, setTransaction] = useState<TransactionStatus>(TransactionStatus.Waiting);
+  const { signTransaction } = useLedger();
+  const { networkName } = useNetwork();
+  const { estimatedFees, errorFees, difference } = useFees(
```

Comment:
Going back at it, as you only sign the TX, there are no fees :) It doesn't go on chain :)